### DISCUSSION TITLE
Equivalent of Sodium's `crypto_box_seal(..)` / `crypto_box_seal_open(..)`?

### DISCUSSION THREAD

**Discussion by getify** [paulmillr/noble-ciphers#32] [General]
I see this quote on the README:

> Nacl / Libsodium popularized term "secretbox", a simple black-box authenticated encryption. Secretbox is just xsalsa20-poly1305. We provide the alias and corresponding seal / open methods.

A couple of questions:

1. Sodium uses the term "secretbox" (`crypto_secretbox_easy(..)` and `crypto_secretbox_easy_open(..)`) for its *symmetric* authenticated encryption (same key for encryption/decryption). Further, they use the term "box" (`crypto_box_easy(..)` and `crypto_box_easy_open(..)`) for their *asymmetric* authenticated encryption (by generating an ephemeral keypair).

    But they use "sealedbox" (`crypto_seal(..)` and `crypto_seal_open(..)`) terminology for its *un-authenticated* (anonymous) *asymmetric* encryption/decryption.

    Do you support (have aliases for) all three of these: "secretbox", "box", and "sealedbox"? Or only some of them?

2. Where specifically on the noble-ciphers API are these various "seal / open method" aliases located? I couldn't find them.

----

My desired goal is to do the "sealedbox" equivalent with Noble, from this Sodium-using code:

```js
let encBuffer = sodium.crypto_seal(msgBuffer, x25519pk, ivBuffer);
let msgBuffer_2 = sodium.crypto_seal_open(encBuffer, x25519sk, ivBuffer);
```

I'm trying to figure out if these aliases already exist, or if the pieces are there in Noble (Ciphers, etc) to do the equivalent operations, specifically where values would be interchangeable between a Sodium-using implementation and a Noble-using implementation.

Any thoughts/tips/guidance are appreciated.

**Comment by paulmillr** [comment]
secretbox is available natively

crypto_box, others etc (combining x25519) are available in noble-sodium wrapper around noble cryptography: https://github.com/serenity-kit/noble-sodium
